test/internal/helpers/Player.java

```diff
@@ -326,6 +327,58 @@ public static Cleanups withClanLoungeItem(final int itemId) {
     return new Cleanups(() -> ClanLoungeRequest.setClanLoungeItem(itemId, 0));
   }
 
+  /**
+   * Restores Clan Furniture after cleanup
+   *
+   * @return Resets to previous state
+   */
+  public static Cleanups withClanFurniture() {
+    var old = ClanManager.getClanRumpus();
+
+    return new Cleanups(
+        () -> {
+          var current = new ArrayList<>(ClanManager.getClanRumpus());
+          for (var item : current) {
+            if (!old.contains(item)) ClanManager.removeFromRumpus(item);
+          }
+        });
+  }
+
+  /**
+   * Adds the given furniture to the player's Clan Rumpus Room
+   *
+   * @param furniture Furniture to install
+   * @return Removes the furniture
+   */
+  public static Cleanups withClanFurniture(final String furniture) {
+    return withClanFurniture(furniture, true);
+  }
+
+  /**
+   * Adds or removes the given furniture to the player's Clan Rumpus Room
+   *
+   * @param furniture Furniture to install
+   * @param shouldHave Whether the player should have the furniture
+   * @return Resets to previous value
+   */
+  public static Cleanups withClanFurniture(final String furniture, final boolean shouldHave) {
```

Comment:
this function is essentially two different functions based on the boolean parameter -- they only share the first line. I think it'd be clearer as `withClanFurniture` and `withoutClanFurniture`.

Reply:
I've never liked withoutX functions but I can change it if you want.

Reply:
I feel like `withoutX` reads much more easily as a condition than `withX(false)`.

In this particular case, I think this function gives a great example of the [flag argument](https://martinfowler.com/bliki/FlagArgument.html) and should be split up for readability.